### ISSUE TITLE
No relevance score when using sorting

### ISSUE THREAD

**Issue by zhaakhi** [vespa-engine/vespa#12163]
We have been using a sorting specification like `-[rank] my_id` to get a consistent ordering in case of ties in the ranking.

This worked fine up to and including version 7.105.45, but from 7.110.36 and up this makes the relevance always come out as 0.

To reproduce, tested in a `vespaengine/vespa:7.175.12` container:
```
mkdir -p app/searchdefinitions
cat <<EOF > app/searchdefinitions/my_documents.sd
search my_documents {
  document my_documents {
        field my_relevance type double {
            indexing: attribute
        }
        field my_id type long {
            indexing: attribute
        }
  }

  rank-profile default {
      first-phase {
          expression: attribute(my_relevance)
      }
  }
}
EOF
cat <<EOF > app/services.xml
<?xml version="1.0" encoding="utf-8" ?>
<services version="1.0">
  <container id="container" version="1.0">
    <document-api />
    <search />
    <nodes>
      <node hostalias="node1" />
    </nodes>
  </container>

  <content id="my_documents" version="1.0">
    <redundancy>1</redundancy>
    <documents>
      <document type="my_documents" mode="index" />
    </documents>
    <nodes>
      <node hostalias="node1" distribution-key="0" />
    </nodes>
    <engine><proton><resource-limits><disk>0.99</disk></resource-limits></proton></engine>
  </content>

</services>
EOF
vespa-deploy prepare app
vespa-deploy activate
sleep 30
curl -s -X POST -H "Content-Type:application/json" --data-binary '{"fields":{"my_id":1,"my_relevance":0.5}}' "http://localhost:8080/document/v1/default/my_documents/docid/1" > /dev/null
curl -s -X POST -H "Content-Type:application/json" --data-binary '{"fields":{"my_id":2,"my_relevance":1.0}}' "http://localhost:8080/document/v1/default/my_documents/docid/2" > /dev/null
curl -s -X POST -H "Content-Type:application/json" --data-binary '{"fields":{"my_id":3,"my_relevance":0.5}}' "http://localhost:8080/document/v1/default/my_documents/docid/3" > /dev/null

# without sorting
curl 'http://localhost:8080/search/?query=my_id:%3E0'  | python -m json.tool
# with sorting
curl 'http://localhost:8080/search/?query=my_id:%3E0&sorting=-\[rank\]%20my_id'  | python -m json.tool
```

The second query is ordered correctly, but all documents are returned with `"relevance": 0.0`

Looking around a bit, one possibly relevant change is the addition of the `LeanHit` class, with one commit labelled [Set relevance to 0.0 if sorting.](https://github.com/vespa-engine/vespa/commit/2e9d2f788e1b891696d215bf0480dbae52233ddb)

As an aside, the [sorting reference](https://docs.vespa.ai/documentation/reference/sorting.html) indicates that `[rank]` is a legacy alias for `[relevance]`, but trying to use it for sorting gives me the error: "The cluster my_documents has no sortable attribute named: [relevance]"

**Comment by jobergum** [comment]
Thank you for the detailed issue description.

**Comment by baldersheim** [comment]
This should now be fixed and will be present in build 7.207.22 and on. The next open source release will be early next week.
